js/ui/search.js

```diff
@@ -377,7 +377,7 @@ const GridSearchResults = new Lang.Class({
     },
 
     _createResultDisplay: function(meta) {
-        return this.parent(meta) || new GridSearchResult(this.provider, meta);
```

Comment:
It strikes me that we may as using `createResultObject()`, which creates an `AppInfo` for the app search provider, is actually a good fit for the applications. Any reason not to style that, instead of changing the upstream behavior?